scripts/doindex/lib/parse.mjs

```diff
@@ -0,0 +1,110 @@
+import { Node } from 'domhandler';
+
+export const DEFAULT_RECORD = Object.freeze({
+    objectID: null,
+    headings: null,
+    mainTitle: null,
+    pageTitle: null,
+    content: null,
+    url: null,
+    type: 'Documentation',
+    parent: null,
+    pageViews: 0,
+    product: 'help/kotlin-reference',
+    pageType: null
+});
+
+/**
+ * @param {string} text
+ * @returns {string}
+ */
+export function cleanText(text) {
+    return text.replace(/\n/g, ' ').trim();
+}
+
+/** @typedef {import('cheerio').CheerioAPI} CheerioAPI */
+
+/**
+ * @param {CheerioAPI} $
+ * @param {Node[]} list
+ * @param {(node: *, level?: number) => boolean} [isFinalNode]
+ * @param {string} [url]
+ * @returns {Promise<string>}
+ */
+export async function htmlToText($, list, isFinalNode, url = '') {
```

Comment:
Looks like `$(...).text()` but with two difference. Wrapping for some nodes and partial parsing node.
It would be nice find some library or solution for this task too.